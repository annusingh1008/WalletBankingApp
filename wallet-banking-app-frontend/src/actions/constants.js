export const authConstants = {
    LOGIN_REQUEST: 'LOGIN_REQUEST',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAILURE: 'LOGOUT_FAILURE'
}

export const userConstants = {
    USER_REGISTER_REQUEST: 'USER_REGISTER_REQUEST',
    USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
    USER_REGISTER_FAILURE: 'USER_REGISTER_FAILURE',
}

export const amountTransferConstants = {
    AMOUNT_TRANSFER_REQUEST: 'AMOUNT_TRANSFER_REQUEST',
    AMOUNT_TRANSFER_SUCCESS: 'AMOUNT_TRANSFER_SUCCESS',
    AMOUNT_TRANSFER_FAILURE: 'AMOUNT_TRANSFER_FAILURE'
}

export const rechargeConstants = {
    RECHARGE_REQUEST: 'RECHARGE_REQUEST',
    RECHARGE_SUCCESS: 'RECHARGE_SUCCESS',
    RECHARGE_FAILURE: 'RECHARGE_FAILURE',
}

export const getUserDetailsConstants = {
    GET_USER_DETAILS_REQUEST: 'GET_USER_DETAILS_REQUEST',
    GET_USER_DETAILS_SUCCESS: 'GET_USER_DETAILS_SUCCESS',
    GET_USER_DETAILS_FAILURE: 'GET_USER_DETAILS_FAILURE'
}

export const transactionsConstants = {
    TRANSACTIONS_REQUEST: 'TRANSACTIONS_REQUEST',
    TRANSACTIONS_SUCCESS: 'TRANSACTIONS_SUCCESS',
    TRANSACTIONS_FAILURE: 'TRANSACTIONS_FAILURE',
    TOTAL_TRANSACTIONS_SUCCESS: 'TOTAL_TRANSACTIONS_SUCCESS',
}

export const cashbackConstants = {
    CASHBACK_REQUEST: 'CASHBACK_REQUEST',
    CASHBACK_SUCCESS: 'CASHBACK_SUCCESS',
    CASHBACK_FAILURE: 'CASHBACK_FAILURE',
    TOTAL_CASHBACKS_SUCCESS: 'TOTAL_CASHBACKS_SUCCESS'
}

export const getBalanceConstants = {
    GET_BALANCE_SUCCESS: 'GET_BALANCE_SUCCESS'
}