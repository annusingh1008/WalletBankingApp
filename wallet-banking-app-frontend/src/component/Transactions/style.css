 .table {
    margin-left: 257px;
    width: 90%;
} 

thead tr:nth-child(1) th {
    background: white;
    position: sticky;
    top: 55px;
}

.no-transactions{
    margin-top: 200px;
    margin-left: 600px;
    font-size: 30px;
    font-weight: bold;
}

.all-btn{
  margin-left: 170px;
}


.button {
  border: 1px solid black;
  background-color: black;
  color: #ffffff;
  padding: 5px 10px;
}

.button:hover {
    border: 1px solid gray;
    background-color: gray;
    color: #ffffff;
    padding: 5px 10px;
  }

.button:disabled,
button[disabled]{
  border: 1px solid #999999;
  background-color: #cccccc;
  color: #666666;
}

.button:focus {     
    background-color: #cccccc;    
    color: black;
}

