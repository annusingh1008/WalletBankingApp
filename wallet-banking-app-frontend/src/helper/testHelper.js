import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const localStorageData = {
  email: "annu",
  password: "1234",
};

export const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    setItem: jest.fn(),
  };
})();

const mockStore = configureMockStore();

export const renderWithRedux = (ui, { store } = {}) => {
  return {
    ...render(<Provider store={mockStore(store)}>{ui}</Provider>),
    store,
  };
};
