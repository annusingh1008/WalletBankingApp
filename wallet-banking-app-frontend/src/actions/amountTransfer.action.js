import axios from "axios";
import { amountTransferConstants } from "./constants";

export const amountTransfer = (amountDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/transfer/amount", {
      ...amountDetails,
    });

    if (res.status === 200) {
      if (res.data === "User does not exist") {
        alert("User does not exist! \nPlease check your email.");
      } else if (res.data === "Amount Transferred Successfully...!!") {
        alert("Transaction Successful");

        dispatch({
          type: amountTransferConstants.AMOUNT_TRANSFER_SUCCESS,
          payload: {
            amount:
              parseInt(amountDetails.currentAmount) -
              parseInt(amountDetails.amount),
            email: amountDetails.email,
          },
        });
      }
    }
  };
};
