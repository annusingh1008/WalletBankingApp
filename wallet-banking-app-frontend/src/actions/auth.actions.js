import axios from "axios";
import { authConstants } from "./constants";

export const login = (user) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/signin", {
      ...user,
    });

    if (res.status === 200) {
      if (res.data === "Signed in Successfully...!!") {
        localStorage.setItem("user", JSON.stringify(user));
        localStorage.setItem("email", user.email);

        dispatch({
          type: authConstants.LOGIN_SUCCESS,
          payload: user,
        });
      } else if (res.data === "Account does not exists with this email...!!") {
        alert("Account does not exists!! \nPlease check your email..");
      } else if (res.data === "Incorrect Password...!!") {
        alert("Please check your Password!!");
      }
    }
  };
};

export const getAuthDetails = (state) => state.auth;

export const signout = () => {
  return async (dispatch) => {
    dispatch({ type: authConstants.LOGOUT_REQUEST });
    localStorage.clear();
    dispatch({ type: authConstants.LOGOUT_SUCCESS });
  };
};
