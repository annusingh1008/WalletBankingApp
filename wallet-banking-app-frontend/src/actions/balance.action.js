import axios from "axios";
import { getBalanceConstants } from "./constants";

export const getUserBalance = (email) => {
  return async (dispatch) => {
    const res = await axios.get(`http://localhost:8085/getBalance/${email}`);

    dispatch({
      type: getBalanceConstants.GET_BALANCE_SUCCESS,
      payload: res.data,
    });
  };
};
