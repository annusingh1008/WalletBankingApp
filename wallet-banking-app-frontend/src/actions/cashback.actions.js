import axios from "axios";
import { cashbackConstants } from "./constants";

export const getAllCashBacks = (currentPage, email) => {
  return async (dispatch) => {
    const userEmail = localStorage.getItem("email");
    const res = await axios.get(
      `http://localhost:8085/getAllCashbacks/${userEmail}?pageNumber=${currentPage}&pageSize=${10}`
    );

    dispatch({
      type: cashbackConstants.CASHBACK_SUCCESS,
      payload: res.data,
    });
  };
};

export const getCashbacks = (state) => state.cashbacks;

export const getTotalCashbacks = (email) => {
  return async (dispatch) => {
    const res = await axios.get(
      `http://localhost:8085/getTotalCashbacks/${email}`
    );

    dispatch({
      type: cashbackConstants.TOTAL_CASHBACKS_SUCCESS,
      payload: res.data,
    });
  };
};
