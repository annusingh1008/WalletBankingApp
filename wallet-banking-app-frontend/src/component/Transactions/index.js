import React, { useEffect, useState } from "react";
import Layout from "../Layout";
import { Table, Button, InputGroup, FormControl, Card } from "react-bootstrap";
import "./style.css";
import { useDispatch, useSelector } from "react-redux";
import {
  getAllTransactions,
  getTotalTransactions,
  getTransactions,
} from "../../actions/transactions.actions";

const Transactions = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getTotalTransactions(email));
    dispatch(getAllTransactions(0, email));
  }, []);

  // const transactions = useSelector((state) => state.transactions);
  const transactions = useSelector(getTransactions);
  const transactionList = transactions?.transactions;
  const email = localStorage.getItem("email");

  const length = transactions?.totalPages;
  const totalPages = length % 10 == 0 ? length / 10 : parseInt(length / 10) + 1;

  const [currentPage, setCurrentPage] = useState(0);

  const firstPage = () => {
    setCurrentPage(0);
    dispatch(getAllTransactions(0, email));
  };

  const prevPage = () => {
    setCurrentPage(currentPage - 1);
    dispatch(getAllTransactions(currentPage - 1, email));
  };

  const lastPage = () => {
    setCurrentPage(totalPages - 1);
    dispatch(getAllTransactions(totalPages - 1, email));
  };

  const nextPage = () => {
    setCurrentPage(currentPage + 1);
    dispatch(getAllTransactions(currentPage + 1, email));
  };

  const convertUTCDateToLocalDate = (date) => {
    var newDate = new Date(
      date.getTime() + date.getTimezoneOffset() * 60 * 1000
    );

    var offset = date.getTimezoneOffset() / 60;
    var hours = date.getHours();

    newDate.setHours(hours - offset);

    return newDate;
  };

  return (
    <Layout sidebar>
      {transactionList?.length > 0 ? (
        <>
          <Table striped bordered hover className="table">
            <thead className="text-center">
              <tr>
                <th scope="col">#</th>
                <th scope="col">From</th>
                <th scope="col">To</th>
                <th scope="col">Credit/ Debit</th>
                <th scope="col">Amount</th>
                <th scope="col">Current Amount</th>
                <th scope="col">Transaction Date</th>
              </tr>
            </thead>
            <tbody className="text-center">
              {transactionList.map((transaction, index) => (
                <tr key={index}>
                  <td>{index + 1}</td>
                  <td>{transaction.from_name}</td>
                  <td>{transaction.to_name}</td>
                  <td>{transaction.type}</td>
                  <td>{transaction.transferAmount}</td>
                  <td>{transaction.amount}</td>
                  <td>
                    {convertUTCDateToLocalDate(
                      new Date(transaction.date)
                    ).toLocaleString()}
                  </td>
                </tr>
              ))}
            </tbody>
          </Table>
          <Card.Footer className="all-btn">
            <div style={{ marginLeft: "450px" }}>
              Showing Page {currentPage + 1} of {totalPages}
            </div>
            <div style={{ marginLeft: "430px" }}>
              <InputGroup size="sm">
                <Button
                  data-testid="first"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === 0 ? true : false}
                  onClick={firstPage}
                >
                  First
                </Button>
                <Button
                  data-testid="prev"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === 0 ? true : false}
                  onClick={prevPage}
                >
                  Prev
                </Button>
                <Button
                  data-testid="next"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === totalPages - 1 ? true : false}
                  onClick={nextPage}
                >
                  Next
                </Button>
                <Button
                  data-testid="last"
                  type="button"
                  className="button"
                  variant="outline-dark"
                  disabled={currentPage === totalPages - 1 ? true : false}
                  onClick={lastPage}
                >
                  Last
                </Button>
              </InputGroup>
            </div>
          </Card.Footer>
        </>
      ) : (
        <div className="no-transactions">No Transactions</div>
      )}
    </Layout>
  );
};

export default Transactions;
