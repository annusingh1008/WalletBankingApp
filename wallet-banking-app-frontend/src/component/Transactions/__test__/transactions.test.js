import React from "react";
import { fireEvent } from "@testing-library/react";
import Transactions from "../index";
import { localStorageMock, renderWithRedux } from "../../../helper/testHelper";
import { getAllTransactions } from "../../../actions/transactions.actions";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../Layout", () =>
  jest.fn(({ children }) => <div data-testid="Layout">{children}</div>)
);

jest.mock("react-bootstrap", () => {
  const Table = jest.fn(({ children }) => (
    <table data-testid="Table">{children}</table>
  ));
  const Card = jest.fn(({ children }) => (
    <div data-testid="Card">{children}</div>
  ));
  Card.Footer = jest.fn(({ children }) => (
    <div data-testid="Footer">{children}</div>
  ));
  const InputGroup = jest.fn(({ children }) => (
    <div data-tesid="InputGroup">{children}</div>
  ));
  const Button = jest.fn((props) => (
    <button
      onClick={props.onClick}
      data-testid={`Button-${props["data-testid"]}`}
    >
      {props.children}
    </button>
  ));

  return { Table, Card, InputGroup, Button };
});

jest.mock("../../../actions/transactions.actions", () => ({
  getTransactions: jest.fn(() => ({
    transactions: [
      {
        email: "abc@gmail.com",
        amount: "1000",
        type: "Credit",
        transferAmount: "1000",
        from_name: "Bank",
        to_name: "Annu Singh",
        date: "2022-10-04T05:50:10.283+00:00",
      },
      {
        email: "abc1@gmail.com",
        amount: "1000",
        type: "Credit",
        transferAmount: "1000",
        from_name: "Bank",
        to_name: "Annu Singh",
        date: "2022-10-04T05:50:10.283+00:00",
      },
      {
        email: "abc2@gmail.com",
        amount: "1000",
        type: "Credit",
        transferAmount: "1000",
        from_name: "Bank",
        to_name: "Annu Singh",
        date: "2022-10-04T05:50:10.283+00:00",
      },
    ],
    totalPages: 10,
  })),
  getTotalTransactions: jest.fn((payload) => ({
    type: "getTotalTransactions",
    payload,
  })),
  getAllTransactions: jest.fn((payload) => ({
    type: "getAllTransactions",
    payload,
  })),
}));

describe("Transactions", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  it("renders properly", () => {
    const { container } = renderWithRedux(<Transactions />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Transactions />);
      const node = queryByTestId("Button-first");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllTransactions(0, "annu"));
    });
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Transactions />);
      const node = queryByTestId("Button-prev");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllTransactions(0, "annu"));
    });
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Transactions />);
      const node = queryByTestId("Button-next");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllTransactions(0, "annu"));
    });
  });

  describe("When clicks on first button", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = renderWithRedux(<Transactions />);
      const node = queryByTestId("Button-last");
      fireEvent.click(node);
      expect(mockDispatch).toBeCalledWith(getAllTransactions(0, "annu"));
    });
  });
});
