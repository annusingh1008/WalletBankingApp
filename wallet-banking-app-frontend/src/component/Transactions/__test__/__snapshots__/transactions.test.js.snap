// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions renders properly 1`] = `
<div
  data-testid="Layout"
>
  <table
    data-testid="Table"
  >
    <thead
      class="text-center"
    >
      <tr>
        <th
          scope="col"
        >
          #
        </th>
        <th
          scope="col"
        >
          From
        </th>
        <th
          scope="col"
        >
          To
        </th>
        <th
          scope="col"
        >
          Credit/ Debit
        </th>
        <th
          scope="col"
        >
          Amount
        </th>
        <th
          scope="col"
        >
          Current Amount
        </th>
        <th
          scope="col"
        >
          Transaction Date
        </th>
      </tr>
    </thead>
    <tbody
      class="text-center"
    >
      <tr>
        <td>
          1
        </td>
        <td>
          Bank
        </td>
        <td>
          Annu Singh
        </td>
        <td>
          Credit
        </td>
        <td>
          1000
        </td>
        <td>
          1000
        </td>
        <td>
          04/10/2022, 16:50:10
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Bank
        </td>
        <td>
          Annu Singh
        </td>
        <td>
          Credit
        </td>
        <td>
          1000
        </td>
        <td>
          1000
        </td>
        <td>
          04/10/2022, 16:50:10
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          Bank
        </td>
        <td>
          Annu Singh
        </td>
        <td>
          Credit
        </td>
        <td>
          1000
        </td>
        <td>
          1000
        </td>
        <td>
          04/10/2022, 16:50:10
        </td>
      </tr>
    </tbody>
  </table>
  <div
    data-testid="Footer"
  >
    <div
      style="margin-left: 450px;"
    >
      Showing Page 
      1
       of 
      1
    </div>
    <div
      style="margin-left: 430px;"
    >
      <div
        data-tesid="InputGroup"
      >
        <button
          data-testid="Button-first"
        >
          First
        </button>
        <button
          data-testid="Button-prev"
        >
          Prev
        </button>
        <button
          data-testid="Button-next"
        >
          Next
        </button>
        <button
          data-testid="Button-last"
        >
          Last
        </button>
      </div>
    </div>
  </div>
</div>
`;
